patches/chrome-browser-ui-views-toolbar-toolbar_view.cc.patch

```diff
@@ -0,0 +1,72 @@
+diff --git a/chrome/browser/ui/views/toolbar/toolbar_view.cc b/chrome/browser/ui/views/toolbar/toolbar_view.cc
+index 3965bc59d1c67af0b84baa052054a5f854904533..ba6f8b6102fb6171d58739eddcda1532af8e6e25 100644
+--- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
++++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
+@@ -69,6 +69,7 @@
+ #include "ui/views/widget/tooltip_manager.h"
+ #include "ui/views/widget/widget.h"
+ #include "ui/views/window/non_client_view.h"
++#include "brave/browser/ui/views/toolbar/bookmark_button.h"
+ 
+ #if defined(OS_WIN) || defined(OS_MACOSX)
+ #include "chrome/browser/recovery/recovery_install_global_error_factory.h"
+@@ -199,6 +200,10 @@ void ToolbarView::Init() {
+   home_->set_id(VIEW_ID_HOME_BUTTON);
+   home_->Init();
+ 
++  bookmark_ = new BookmarkButton(browser_->profile(), this);
++  bookmark_->set_triggerable_event_flags(ui::EF_LEFT_MOUSE_BUTTON | ui::EF_MIDDLE_MOUSE_BUTTON);
++  bookmark_->Init();
++
+   // No master container for this one (it is master).
+   BrowserActionsContainer* main_container = nullptr;
+   browser_actions_ =
+@@ -228,6 +233,7 @@ void ToolbarView::Init() {
+   AddChildView(forward_);
+   AddChildView(reload_);
+   AddChildView(home_);
++  AddChildView(bookmark_);
+   AddChildView(location_bar_);
+   AddChildView(browser_actions_);
+   if (avatar_)
+@@ -308,6 +314,7 @@ void ToolbarView::ShowBookmarkBubble(
+     bool already_bookmarked,
+     bookmarks::BookmarkBubbleObserver* observer) {
+   views::View* anchor_view = location_bar();
++#if !defined(BRAVE_CHROMIUM_BUILD)
+   PageActionIconView* const star_view = location_bar()->star_view();
+   if (!ui::MaterialDesignController::IsSecondaryUiMaterial()) {
```

Comment:
this seems to be controlled by IsSecondaryUiMaterial which I believe is controllable by a flag

Reply:
@bridiver yes and that flag is true by default at the moment as far as I know, so this code block shouldn't get run anyway. However, we still need the `PageActionIconView* const star_view = location_bar()->star_view();` line not to run. However, if we follow your suggestion to replace `location_bar` with a class that encompasses the `BookmarkButton` and the actual `LocationBarView` then this may not be necessary.